pkg/nuctl/command/redeploy.go

```diff
@@ -0,0 +1,356 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/common/headers"
+	"github.com/nuclio/nuclio/pkg/errgroup"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	nuctlcommon "github.com/nuclio/nuclio/pkg/nuctl/command/common"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/nuclio-sdk-go"
+	"github.com/spf13/cobra"
+)
+
+type redeployCommandeer struct {
+	cmd            *cobra.Command
+	rootCommandeer *RootCommandeer
+
+	betaCommandeer         *betaCommandeer
+	redeployFromReportFile bool
+	deployAll              bool
+	waitForFunction        bool
+	skipSpecCleanup        bool
+	verifyExternalRegistry bool
+	outputManifest         *nuctlcommon.PatchManifest
+	inputManifest          *nuctlcommon.PatchManifest
+	saveReport             bool
+	reportFilePath         string
+	excludedProjects       []string
+	excludedFunctions      []string
+	excludeFunctionWithGPU bool
+	importedOnly           bool
+	waitTimeout            time.Duration
+}
+
+func newRedeployCommandeer(ctx context.Context, rootCommandeer *RootCommandeer, betaCommandeer *betaCommandeer) *redeployCommandeer {
+	commandeer := &redeployCommandeer{
+		rootCommandeer: rootCommandeer,
+		outputManifest: nuctlcommon.NewPatchManifest(),
+		betaCommandeer: betaCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "redeploy function-name",
+		Short: "Redeploy a function",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// initialize root
+			if err := rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			if commandeer.betaCommandeer != nil {
+				if err := commandeer.betaCommandeer.initialize(); err != nil {
+					return errors.Wrap(err, "Failed to initialize beta commandeer")
+				}
+				commandeer.rootCommandeer.loggerInstance.Debug("In BETA mode")
+				if err := commandeer.betaRedeploy(ctx, args); err != nil {
+					return errors.Wrap(err, "Failed to deploy function")
+				}
+				return nil
+			}
+			return nil
+		},
+	}
+
+	addRedeployFlags(cmd, commandeer)
+	cmd.Flags().BoolVarP(&commandeer.skipSpecCleanup, "skip-spec-cleanup", "", false, "Do not clean up spec in function configs")
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func addRedeployFlags(cmd *cobra.Command,
+	commandeer *redeployCommandeer) {
+	cmd.Flags().BoolVar(&commandeer.deployAll, "deploy-all", false, "Deploy all functions in the namespace")
```

Comment:
This is inferred if `len(args) == 0`, and is not used anywhere anyway - you can delete it :)